doc/specs/#653 - Quake Mode/#653 - Quake Mode.md

```diff
@@ -0,0 +1,629 @@
+
+else
+* Create a new window on this monitor  & desktop
+</td>
+<td>
+
+If there is a window on monitor N on any desktop,
+* Move the window to this desktop (leave position alone)
+
+else
+* Create a new window on this monitor & desktop
+</td>
+<td>
+
+If there isn't one on this desktop, (even if there is one on monitor N on
+another desktop),
+* create a new one on monitor N
+
+Else if ( there is one on this desktop, not monitor N)
+* create a new one on monitor N
+
+Else (one on this desktop & monitor N)
+* Activate the one on this desktop (don't move)
+</td>
+</tr>
+</table>
+
+
+##### Stories, revisited
+
+With the above settings, let's re-examine the original user stories, and see how
+they fit into the above settings. (_Stories that are omitted aren't relevant to
+the discussion of these settings_)
+
+> When the `desktop` param is omitted below, that can be interpreted as "any
+> `desktop` value will make sense here"
+
+* **Story A** Press a hotkey anywhere to activate the single Terminal window
+  wherever it was
+  - This is `{ "monitor": "any", "desktop": null }`
+* **Story B** Press a hotkey anywhere to activate the single Terminal window _on
+  the current monitor_. If it wasn't previously on that monitor, move it there.
+  - This is `{ "monitor": "toCurrent" }`
+* **Story D** <kbd>Ctrl+1</kbd> to activate the terminal on monitor 1,
+  <kbd>Ctrl+2</kbd> to activate the terminal on monitor 2.
+  - This is `[ { "keys": "ctrl+1", monitor": 1 }, { "keys": "ctrl+2", monitor": 2 } ]`
+
+As some additional examples:
+
+```json
+// Go to the MRU window, wherever it is
+{ "keys": "win+1", "monitor":"any", "desktop": null },
+// Since "any" & null are the default values, just placing a single entry here
+// will bind the same behavior:
+{ "keys": "win+1" },
+
+// activate the MRU window, and move it to this desktop & this monitor
+{ "keys": "win+2", "monitor":"toCurrent", "desktop": "toCurrent" },
+
+// activate the MRU window on this desktop
+{ "keys": "win+3", "monitor":"any", "desktop": "onCurrent" },
+
+// Activate the MRU window on monitor 2 (from any desktop), and place it on the
+// current desktop. If there isn't one on monitor 2, make a new one.
+{ "keys": "win+4", "monitor": 2, "desktop": "toCurrent" },
+
+// Activate the MRU window on monitor 3 (ONLY THIS desktop), or make a new one.
+{ "keys": "win+5", "monitor": 3, "desktop": "onCurrent" },
+
+// Activate the MRU window on this monitor (from any desktop), and place it on
+// the current desktop. If there isn't one on this monitor, make a new one.
+{ "keys": "win+6", "monitor": "onCurrent", "desktop": "toCurrent" },
+```
+
+#### Other properties
+
+Some users would like the terminal to just appear when the global hotkey is
+pressed. Others would like the true quake-like experience, where the terminal
+window "slides-in" from the top of the monitor. Furthermore, some users would
+like to configure the speed at which that dropdown happens. To support this

```

Comment:
>  Furthermore, some users would like to configure the speed at which that dropdown happens.

What other terminals let you customize this? It looks like iTerm2 doesn't have this. This is another setting that I feel may be a bit too customizable.

I'd feel better if we are able to simplify it to a boolean. That said, we could just expose it as a toggle button on the Settings UI and do some translation between the view model and the settings model. But still.

Reply:
there was a specific request for this, so I'm gonna leave it in.